cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -3186,9 +3180,19 @@ class AzureFileSystem::Impl {
       }
     }
     try {
-      dest_blob_client.CopyFromUri(src_url);
+      // We use StartCopyFromUri instead of CopyFromUri because it supports blobs larger
+      // than 256 MiB and it doesn't require generating a SAS token to authenticate
+      // reading a source blob in the same storage account.
+      auto copy_operation = dest_blob_client.StartCopyFromUri(src_url);
```

Comment:
Wow! I should have found it when I implement this...

Reply:
I was quite shocked to find the 256MiB limit, to the point where I initially assumed the documentation was out of date, but I tried it and it really did return an error. I guess its a policy to avoid a single slow API call.